hw/top_earlgrey/data/ip/chip_i2c_testplan.hjson

```diff
@@ -38,9 +43,52 @@
             - Verify the virtual / true open drain capability.
 
             Verify all instances of I2C in the chip.
+
+            Notes for silicon targets:
+            - The testbench in this case will be a device that physically toggles/inspects the appropriate SCL and SDA lines.
             '''
+      features: ["I2C.MODE.TARGET", "I2C.OPERATION.READ", "I2C.OPERATION.WRITE"]
       stage: V2
+      si_stage: SV2
       tests: ["chip_sw_i2c_device_tx_rx"]
     }
+    {
+      name: chip_sw_i2c_speed
+      desc: '''Verify the transmission speed of data over the chip's I2C interfaces.
+
+            Run the following for standard, fast and fast plus speeds:
+            - Program the I2C to be in the appropriate mode (host/target).
+            - For hosts, the SW test writes a known payload over the chip's I2C host interface.
```

Comment:
I suggest a loopback test where the host writes X and reads X+1 or X>>1.
Should we describe here how the baud rate will be measured (Since i2c is synchronous, the fact the data matches doesn't mean the speed is correct)?

Reply:
Thanks for this suggestion. I've made an initial attempt at this, but happy to adjust it based on further feedback.